sequencer/src/api.rs

```diff
@@ -414,6 +424,21 @@ pub mod test_helpers {
             Self { server, peers, cfg }
         }
 
+        pub async fn new_with_config(
```

Comment:
Builder pattern is a good suggestion but it would be a overhaul. I made this change because I think the current constructors are not good enough because we're required to pass the fields that differ from the default values  to the constructor.
I prefer running a binary by passing the config. It looks straightforward and the interface is clean. I believe passing through the `config` to run the binary is the best practice

Reply:
I don't disagree at all, but I still think we need to clean this up somehow. Maybe `TestNetwork::new()` should just require a config? If these changes are too invasive for this PR, then maybe the solution is documentation as in the other comments. We can also add some TODOS.

Reply:
https://github.com/EspressoSystems/espresso-sequencer/pull/1583/commits/38fa28835fdb41e1ddaf7af6f1db35887284e45d